main.tf

```diff
@@ -50,14 +44,12 @@ resource "aws_vpn_connection" "this" {
 }
 
 resource "aws_vpn_connection_route" "this" {
-  count = var.create_vpn_connection ? length(var.destination_cidr_blocks) : 0
 
   destination_cidr_block = var.destination_cidr_blocks[count.index]
   vpn_connection_id      = aws_vpn_connection.this[0].id
 }
 
 resource "aws_vpn_gateway_route_propagation" "this" {
-  count = var.create_vpn_connection ? var.propagating_route_table_count : 0
```

Comment:
Still need to count over var.propagating_route_table_count

Reply:
`count = var.propagating_route_table_count` ?